src/main/java/jenkins/branch/buildstrategies/basic/AllBranchBuildStrategyImpl.java

```diff
@@ -79,7 +98,8 @@ public boolean isAutomaticBuild(@NonNull SCMSource source, @NonNull SCMHead head
                 head,
                 currRevision,
                 prevRevision,
-                taskListener
+                taskListener,
+                lastSeenRevision
```

Comment:
this won't help with https://issues.jenkins-ci.org/browse/JENKINS-56984 it makes it even further divergent.

Reply:
That's because the strategy is breaking the API->SPI->API->SPI chain and going direct from SPI->SPI the method being invoked *should* be

```
        for (BranchBuildStrategy strategy: strategies) {
            if(!strategy.automaticBuild(
                source,
                head,
                currRevision,
                prevRevision,
```